src/with-urql-client.tsx

```diff
@@ -38,14 +44,19 @@ const withUrqlClient = <T extends {}>(
   App: NextComponentClass<T & WithUrqlClient> | NextFC<T & WithUrqlClient>,
 ) => {
   const withUrql: NextFC<
-    T & WithUrqlClient & WithUrqlState,
+    T & WithUrqlClient & WithUrqlState & { ctx: NextContextWithAppTree },
     T & WithUrqlState,
     NextContextWithAppTree
   > = props => {
+    const opts =
+      typeof clientOptions === 'function'
```

Comment:
Not sure if other folks have better // more efficient // safer ways for determining whether or not a value is of type `'function'`. I saw a few other implementations on Stack Overflow but they all felt fairly heavy-handed 🤔